src/js/graphql_endpoint/modules/root_query.js

```diff
@@ -0,0 +1,266 @@
+							filterDefaultDgraphNodeTypes
+						);
+					}
+				});
+
+				// filter out nodes that don't have dgraph_types
+				neighbor[predicate] = neighbor[predicate].filter(
+					(node) => "dgraph_type" in node && !!node["dgraph_type"]
+				);
+				continue;
+			}
+
+			// If this edge is 1-to-many, we need to filter down the list to lens-neighbor -> lens-neighbor connections
+			if (
+				Array.isArray(neighbor[predicate]) &&
+				neighbor[predicate] &&
+				neighbor[predicate][0]["uid"]
+			) {
+				neighbor[predicate].forEach(
+					(node) => (node["uid"] = parseInt(node["uid"], 16))
+				);
+				neighbor[predicate].forEach(
+					(node) =>
+						(node["dgraph_type"] = node["dgraph_type"].filter(
+							filterDefaultDgraphNodeTypes
+						))
+				);
+				neighbor[predicate] = neighbor[predicate].filter((second_neighbor) =>
+					neighbor_uids.has(second_neighbor["uid"])
+				);
+
+				// If we filtered all the edges down, might as well delete this predicate
+				if (neighbor[predicate].length === 0) {
+					delete neighbor[predicate];
+				}
+			}
+			// If this edge is 1-to-1, we need to determine if we need to delete the edge
+			else if (
+				typeof neighbor[predicate] === "object" &&
+				neighbor[predicate]["uid"]
+			) {
+				if (!neighbor_uids.has(parseInt(neighbor[predicate]["uid"], 16))) {
+					delete neighbor[predicate];
+				} else {
+					neighbor[predicate]["uid"] = parseInt(neighbor[predicate]["uid"], 16);
+					neighbor[predicate]["dgraph_type"] = neighbor[predicate][
+						"dgraph_type"
+					].filter(filterDefaultDgraphNodeTypes);
+				}
+			}
+		}
+	}
+
+	for (node of lens_subgraph["scope"]) {
+		if (!builtins.has(node.dgraph_type[0])) {
+			const tmpNode = { ...node };
+			node.predicates = tmpNode;
+		}
+	}
+
+	console.log("lens_subgraph scope", JSON.stringify(lens_subgraph["scope"]));
+	return lens_subgraph;
+};
+
+
+const RootQuery = new GraphQLObjectType({
+	name: "RootQueryType",
+	fields: {
+		lenses: {
+			type: GraphQLList(LensNodeType),
+			args: {
+				first: {
+					type: new GraphQLNonNull(GraphQLInt),
+				},
+				offset: {
+					type: new GraphQLNonNull(GraphQLInt),
+				},
+			},
+			resolve: async (parent, args) => {
+				console.log("lenses query arguments", args);
+				const first = args.first;
+				const offset = args.offset;
+				// #TODO: Make sure to validate that 'first' is under a specific limit, maybe 1000
+				console.log("Making getLensesQuery");
+				const lenses = await getLenses(getDgraphClient(), first, offset);
+				console.debug("returning data from getLenses for lenses resolver", lenses);
+				return lenses;
+			},
+		},
+		lens_scope: {
+			type: LensNodeType,
+			args: {
+				lens_name: { type: new GraphQLNonNull(GraphQLString) },
+			},
+			resolve: async (parent, args) => {
+				try {
+					console.log("lens_scope args: ", args);
+					let response = await handleLensScope(parent, args);
+					console.log("lens_scope response: ", response);
+					return response;
```

Comment:
This is the only code change; originally it called `handleLensScope` again instead of returning the previously stored response